.github/workflows/test-coverage.yml

```
@@ -0,0 +1,45 @@
+          fetch-depth: 0 # Fetch full history for SonarQube branch analysis
+
+      # Set up JDK (match your project’s Java version, e.g., 17)
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      # Cache Gradle dependencies
+      - name: Cache Gradle packages
+        uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)